addons/mail/static/src/models/link_preview_list_view.js

```diff
@@ -0,0 +1,125 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, many, one } from '@mail/model/model_field';
+import { insertAndReplace, replace } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'LinkPreviewListView',
+    identifyingFields: ['messageViewOwner'],
+    recordMethods: {
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeLinkPreviewAsCardViews() {
+            return insertAndReplace(
+                this.messageViewOwner.message.linkPreviews
+                    .filter(linkPreview => linkPreview.isCard)
+                    .map(linkPreview => ({ linkPreview: replace(linkPreview) }))
+            );
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeLinkPreviewAsImageViews() {
+            return insertAndReplace(
+                this.messageViewOwner.message.linkPreviews
+                    .filter(linkPreview => linkPreview.isImage)
+                    .map(linkPreview => ({ linkPreview: replace(linkPreview) }))
+            );
+        },
+        /**
+         * @private
+         * @returns {FieldCommand}
+         */
+        _computeLinkPreviewAsVideoViews() {
+            return insertAndReplace(
+                this.messageViewOwner.message.linkPreviews
+                    .filter(linkPreview => linkPreview.isVideo)
+                    .map(linkPreview => ({ linkPreview: replace(linkPreview) }))
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsInChatWindow() {
+            return Boolean(this.messageViewOwner.isInChatWindow);
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsInChatWindowAndIsAlignedRight() {
+            return Boolean(
+                this.isInChatWindow &&
+                this.isCurrentUserOrGuestAuthor
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsInChatWindowAndIsAlignedLeft() {
+            return Boolean(
+                this.isInChatWindow &&
+                !this.isCurrentUserOrGuestAuthor
+            );
+        },
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsCurrentUserOrGuestAuthor() {
+            return Boolean(this.messageViewOwner.message.isCurrentUserOrGuestAuthor);
+        },
```

Comment:
should check author of the preview instead?

Reply:
I've posted the comment several times but actually now that message is required we could remove guest_id from the preview instead and actually use message like you did?